### ISSUE TITLE
Are dimensions supposed to be integers?

### ISSUE THREAD

**Issue by beausmith** [Sunhat/react-native-extra-dimensions-android#19]
I created a brand new app using React Native version 0.44.0, followed the install steps for `react-native-extra-dimensions-android`, and added the following code to the `index.android.js`:

    import React, { Component } from 'react';
    import {
      AppRegistry,
      Dimensions,
      Text,
      View
    } from 'react-native';
    import ExtraDimensions from 'react-native-extra-dimensions-android';

    export default class HelloWorld extends Component {
      render() {
        return (
          <View style={{flex: 1, justifyContent: 'center', alignItems: 'center', backgroundColor: '#F5FCFF'}}>
            <Text style={{fontSize: 20, textAlign: 'center', margin: 10}}>Welcome to React Native!</Text>
            <Text>To get started, edit index.android.js</Text>
            <View style={{margin: 20}}>
              <Text style={{marginTop: 20, fontWeight: 'bold'}}>Dimensions.get('screen')</Text>
              <Text>width: {Dimensions.get('screen').width}</Text>
              <Text>height: {Dimensions.get('screen').height}</Text>
              <Text style={{marginTop: 20, fontWeight: 'bold'}}>Dimensions.get('window')</Text>
              <Text>width: {Dimensions.get('window').width}</Text>
              <Text>height: {Dimensions.get('window').height}</Text>
              <Text style={{marginTop: 20, fontWeight: 'bold'}}>ExtraDimensions</Text>
              <Text>REAL_WINDOW_HEIGHT: {ExtraDimensions.get('REAL_WINDOW_HEIGHT')}</Text>
              <Text>REAL_WINDOW_WIDTH: {ExtraDimensions.get('REAL_WINDOW_WIDTH')}</Text>
              <Text>STATUS_BAR_HEIGHT: {ExtraDimensions.get('STATUS_BAR_HEIGHT')}</Text>
              <Text>SOFT_MENU_BAR_HEIGHT: {ExtraDimensions.get('SOFT_MENU_BAR_HEIGHT')}</Text>
              <Text>SMART_BAR_HEIGHT: {ExtraDimensions.get('SMART_BAR_HEIGHT')}</Text>
            </View>
          </View>
        );
      }
    }

    AppRegistry.registerComponent('HelloWorld', () => HelloWorld);

This is what renders to the screen:

![image](https://cloud.githubusercontent.com/assets/28444/26187203/fb9fa734-3b4b-11e7-8de8-a6acd5379418.png)

I expected that the `react-native-extra-dimensions-android` values to be whole integers. 

If I visit <http://whatismyscreenresolution.com/> I see that the screen resolution is: 412 x 732

![image](https://cloud.githubusercontent.com/assets/28444/26215490/62200990-3bb5-11e7-9230-3a932105973e.png)

Am I doing something wrong?

**Comment by Sunhat** [comment]
This is an unusual bug. I can look at fixing this. I could look into the Java that pulls this information, or I could just round off the numbers in the JS.

My concern is, in the case you've presented, it would need to be rounded up, and perhaps in another case, it could need to be rounded down?

**Comment by Sunhat** [comment]
Thanks for letting me know about this. Sorry for the delay. Fix incoming.

**Comment by Sunhat** [comment]
Only two years later...

**Comment by beausmith** [comment]
Thanks!!

**Comment by Sunhat** [comment]
I've updated the code to revert this.

After initially assuming this was a bug from the previous developer who made this project, I decided to look into this further.

The resulting dimensions of width/height are not pixel calculations.

Here's how the calculations work:

Each device has a set DPI. Usually a multiple of 160. Pixel XL and some others are an exception.  I'll use the Pixel XL as an example, as it's what I've been using to test on.

160dpi is a standard low-res display. I have no idea why 160 was chosen. That's just how it's advised. If you're making graphics for Android, it's advised to make images for 160*1,160*2, 160*3, etc displays.

The device is 2560h x 1440w.
The device DPI is 560.
560 / 160 = 3.5. A bit awkward.

The window height:
2560 / 3.5 = 731.4285714286
This is not a PX value, but a value that will scale to fix the 2560px height display.


The result is acceptable. Rounding could cause problems with slightly larger or smaller than desired measurements.
